linux/bindings/src/crypto/skcipher.rs

```diff
@@ -0,0 +1,264 @@
+/// ```
+#[pin_data(PinnedDrop)]
+pub struct Skcipher {
+    #[pin]
+    write_access: Mutex<()>,
+    skcipher: NonNull<crate::crypto_skcipher>,
+}
+
+// SAFETY: `skcipher` is pointer to `crypto_skcipher`, which could be safely
+// transferred across threads.
+unsafe impl Send for Skcipher {}
+
+// SAFETY: `write_access` ensure the synchronization, so it could be
+// used concurrently from multiple threads.
+unsafe impl Sync for Skcipher {}
+
+impl Skcipher {
+    /// Constructs a new initializer.
+    ///
+    /// Try to allocate an skcipher handler, `alg_name` specifies the driver
+    /// name of the cipher, `type_` specifies the type of the cipher, `mask`
+    /// specifies the mask for the cipher.
+    ///
+    /// Returns a type impl [`PinInit<Skcipher>`] in case of success, or an error
+    /// code in [`Error`].
+    pub fn new(alg_name: &'static CStr, type_: u32, mask: u32) -> impl PinInit<Self, Error> {
+        try_pin_init!(Self {
+            write_access <- new_mutex!(()),
+            // SAFETY: `alg_name` has static lifetimes and live indefinitely.
+            // Any error will be checked by `IS_ERR`.
+            skcipher: unsafe {
+                let skcipher = crate::crypto_alloc_skcipher(alg_name.as_char_ptr(), type_, mask);
+                let const_ptr: *const core::ffi::c_void = skcipher.cast();
+                if IS_ERR(const_ptr) {
+                    let err = PTR_ERR(const_ptr) as i32;
+                    return Err(to_result(err).err().unwrap());
+                }
+                NonNull::new_unchecked(skcipher)
+            },
+        })
+    }
+
+    /// Allocates a cipher request, which could be used to handle the cipher
+    /// operations later.
+    pub fn alloc_request(&self) -> Result<SkcipherReq> {
+        // SAFETY: `self.skcipher` is a valid cipher.
+        let req = unsafe { crate::skcipher_request_alloc(self.skcipher.as_ptr(), GFP_KERNEL) };
+        if req.is_null() {
+            return Err(ENOMEM);
+        }
+
+        Ok(SkcipherReq {
+            // SAFETY: `req` is checked above, it is valid.
+            req: unsafe { NonNull::new_unchecked(req) },
+        })
+    }
+
+    /// Returns the size (in bytes) of the IV for the skcipher referenced by the
+    /// cipher handle. This IV size may be zero if the cipher does not need an IV.
+    pub fn ivsize(&self) -> usize {
+        // SAFETY: `self.skcipher` is constructed by `new`, it is non-null and valid.
+        unsafe {
+            let crypto_alg = (*self.skcipher.as_ptr()).base.__crt_alg;
+            let skcipher_alg = crate::container_of!(crypto_alg, crate::skcipher_alg, base);
+            (*skcipher_alg).ivsize as _
+        }
+    }
+
+    /// Returns the size (in bytes) of the request data structure. The `skcipher_request`
+    /// data structure contains all pointers to data required for the skcipher operation.
+    pub fn reqsize(&self) -> usize {
```

Comment:
Why the sizes here not Rust code style? (no underline)

Reply:
To avoid confusion when switching back and forth between the C and Rust sides.